daemon/daemon.cc

```diff
@@ -944,28 +985,25 @@ static bool init_daemon_data(int argc, char* argv[], struct daemon_data* data, b
 
     if (dumpSettings)
     {
-        auto const str = tr_variantToStr(&data->settings, TR_VARIANT_FMT_JSON);
+        auto const str = tr_variantToStr(&settings, TR_VARIANT_FMT_JSON);
         fprintf(stderr, "%s", str.c_str());
         goto EXIT_EARLY;
     }
 
     return true;
 
 EXIT_EARLY:
-    tr_variantClear(&data->settings);
+    tr_variantClear(&settings);
     return false;
 }
 
 int tr_main(int argc, char* argv[])
 {
-    key_pidfile = tr_quark_new("pidfile"sv);
-    key_watch_dir_force_generic = tr_quark_new("watch-dir-force-generic"sv);
-
-    struct daemon_data data;
+    Daemon daemon;
     bool foreground;
     int ret;
 
-    if (!init_daemon_data(argc, argv, &data, &foreground, &ret))
+    if (!daemon.init(argc, argv, &foreground, &ret))
```

Comment:
Maybe move the Daemon declaration from 1002 to 1006 and replace `init()` with a constructor that takes argc + argv + foreground + ret?

Reply:
Not sure. Since constructors can't have the return value, this assumes an extra boolean field to indicate whether an initialization was succeeded. Otherwise we're forced to throw an exception from the constructor itself, which is, well, somewhat confusing at least.